src/orchestrator-kubernetes/src/lib.rs

```diff
@@ -1193,32 +1197,32 @@ impl NamespacedOrchestrator for NamespacedKubernetesOrchestrator {
                 pod_template_hash.clone(),
             );
 
-        let stateful_set = StatefulSet {
-            metadata: ObjectMeta {
-                name: Some(name.clone()),
-                ..Default::default()
-            },
-            spec: Some(StatefulSetSpec {
-                selector: LabelSelector {
-                    match_labels: Some(match_labels),
+        let stateful_sets = process_names
+            .iter()
+            .map(|process_name| StatefulSet {
+                metadata: ObjectMeta {
+                    name: Some(process_name.clone()),
+                    labels: Some(match_labels.clone()),
                     ..Default::default()
                 },
-                service_name: name.clone(),
-                replicas: Some(scale.into()),
-                template: pod_template_spec,
-                pod_management_policy: Some("Parallel".to_string()),
-                volume_claim_templates,
-                ..Default::default()
-            }),
-            status: None,
-        };
+                spec: Some(StatefulSetSpec {
+                    selector: LabelSelector {
+                        match_labels: Some(match_labels.clone()),
+                        ..Default::default()
+                    },
```

Comment:
I'm unsure of this part: The pod selector for each StatefulSet also selects the pods from other StatefulSets, which seems like it would be a problem, but things seem to work anyway.